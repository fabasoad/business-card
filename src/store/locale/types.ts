export const LOAD_LOCALE = 'LOAD_LOCALE'
export const SET_LOCALE = 'SET_LOCALE'

export class Locale {
  code: string
  title: string

  constructor(code: string, title: string = null) {
    this.code = code
    this.title = ` ${title || code}`
  }
}

interface LoadLocaleAction {
  type: typeof LOAD_LOCALE
  payload: Locale
}

interface SetLocaleAction {
  type: typeof SET_LOCALE
  payload: Locale
}

export type LocaleActionTypes = LoadLocaleAction | SetLocaleAction

export type AppActions = LocaleActionTypes
