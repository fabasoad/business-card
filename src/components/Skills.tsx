/// <reference path='./Skills.d.ts' />
import * as React from 'react'
import { useTranslation } from 'react-i18next'
import { v4 as uuidv4 } from 'uuid'

import TechnologyStorage from '../scripts/technologies/TechnologyStorage'

interface SkillsProps {
  technologyStorage?: TechnologyStorage
}

export default function Skills(props: SkillsProps) {
  const { t } = useTranslation()
  return (
    <div id="skills" className="light-component">
      <div className="section-title text-center center">
        <h2>{t('business-card-skills-title')}</h2>
        <hr />
      </div>
      <div className="container">
        <div className="row">
          <div className="skills-list col text-center">
            {props.technologyStorage.findBySkill(true).map((t) => <img key={uuidv4()} className="m-4" src={t.img} alt={t.name} title={t.title} />)}
          </div>
        </div>
      </div>
    </div>
  )
}

Skills.defaultProps = {
  technologyStorage: new TechnologyStorage()
}
