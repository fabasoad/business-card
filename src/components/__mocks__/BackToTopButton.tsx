import * as React from 'react'

export default function BackToTopButton() {
  return (
    <div data-testid="BackToTopButton"></div>
  )
}
