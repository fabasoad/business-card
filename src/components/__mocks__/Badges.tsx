import * as React from 'react'

export default function Badges() {
  return (
    <div data-testid="Badges"></div>
  )
}
