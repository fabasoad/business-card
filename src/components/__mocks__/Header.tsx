import * as React from 'react'

export default function Header() {
  return (
    <div data-testid="Header"></div>
  )
}
