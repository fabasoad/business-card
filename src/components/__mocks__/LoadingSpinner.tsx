import * as React from 'react'
import { LoadingSpinnerProps } from '../LoadingSpinner'

export default function LoadingSpinner({ background, color }: LoadingSpinnerProps) {
  return (
    <div data-testid={`LoadingSpinner-${background}-${color}`}></div>
  )
}
