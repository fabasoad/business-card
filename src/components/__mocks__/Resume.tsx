import * as React from 'react'

export default function Resume() {
  return (
    <div data-testid="Resume"></div>
  )
}
