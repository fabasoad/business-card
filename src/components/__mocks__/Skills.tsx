import * as React from 'react'

export default function Skills() {
  return (
    <div data-testid="Skills"></div>
  )
}
