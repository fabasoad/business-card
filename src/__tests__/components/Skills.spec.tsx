/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import Skills from '../../components/Skills'
import { useTranslation } from '../__mocks__/react-i18next'
import { Technology } from '../../scripts/technologies/types'
import TechnologyStorage from '../../scripts/technologies/TechnologyStorage'

let tMock

beforeAll(() => {
  tMock = useTranslation().t
})

test('should render Skills correctly', () => {
  const findBySkillSpy: () => Array<Technology> = jest.fn(
    () => new Array<number>(1, 2).map(
      (n) => ({ name: `name-${n}`, title: `title-${n}`, img: `img-${n}`, isSkill: true })
    )
  )
  const technologyStorage = new TechnologyStorage(null)
  technologyStorage.findBySkill = findBySkillSpy
  const wrapper: ShallowWrapper = shallow(
    <Skills technologyStorage={technologyStorage} />
  )
  expect(wrapper).toMatchSnapshot()
  expect(tMock).toHaveBeenCalledTimes(1)
  expect(tMock).toHaveBeenCalledWith('business-card-skills-title')
  expect(findBySkillSpy).toHaveBeenCalledTimes(1)
  expect(findBySkillSpy).toHaveBeenCalledWith(true)
})
