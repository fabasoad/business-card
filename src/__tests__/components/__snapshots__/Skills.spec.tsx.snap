// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Skills correctly 1`] = `
<div
  className="light-component"
  id="skills"
>
  <div
    className="section-title text-center center"
  >
    <h2>
      business-card-skills-title
    </h2>
    <hr />
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="skills-list col text-center"
      >
        <img
          alt="name-1"
          className="m-4"
          key="test-uuid"
          src="img-1"
          title="title-1"
        />
        <img
          alt="name-2"
          className="m-4"
          key="test-uuid"
          src="img-2"
          title="title-2"
        />
      </div>
    </div>
  </div>
</div>
`;
